src/musicbrainz/musicbrainzxml.cpp

```diff
@@ -0,0 +1,508 @@
+#include "musicbrainz/musicbrainzxml.h"
+
+#include <QTextStream>
+#include <QXmlStreamReader>
+#include <QTextCodec>
+
+#include "util/logger.h"
+
+
+namespace mixxx {
+
+namespace musicbrainz {
+
+namespace {
+
+Logger kLogger("MusicBrainz XML");
+
+constexpr int kDefaultErrorCode = 0;
+
+bool finishElement(
+        const QXmlStreamReader& reader,
+        const QString& elementName = QString()) {
+    if (reader.tokenType() != QXmlStreamReader::EndElement) {
+        return false;
+    }
+    VERIFY_OR_DEBUG_ASSERT(elementName.isEmpty() || reader.name() == elementName) {
+        kLogger.warning()
+                << "Unexpected end element tag"
+                << reader.name()
+                << "instead of"
+                << elementName;
+        return false;
+    }
+    return true;
+}
+
+bool continueReading(QXmlStreamReader& reader) {
+    return !(reader.atEnd() || reader.hasError());
+}
+
+void consumeCurrentElement(QXmlStreamReader& reader) {
+    DEBUG_ASSERT(reader.tokenType() == QXmlStreamReader::StartElement);
+    const auto elementName = reader.name().toString();
+    int level = 0;
+    while (continueReading(reader)) {
+        switch (reader.readNext()) {
+        case QXmlStreamReader::StartElement:
+            if (kLogger.traceEnabled()) {
+                kLogger.trace()
+                        << "Start skipping"
+                        << reader.name();
+            }
+            ++level;
+            break;
+        case QXmlStreamReader::EndElement:
+            if (level == 0) {
+                finishElement(reader, elementName);
+                return;
+            }
+            DEBUG_ASSERT(level > 0);
+            if (kLogger.traceEnabled()) {
+                kLogger.trace()
+                        << "End skipping"
+                        << reader.name();
+            }
+            --level;
+            break;
+        default:
+            DEBUG_ASSERT(reader.tokenType() != QXmlStreamReader::NoToken);
+            DEBUG_ASSERT(reader.tokenType() != QXmlStreamReader::Invalid);
+            DEBUG_ASSERT(reader.tokenType() != QXmlStreamReader::StartDocument);
+            DEBUG_ASSERT(reader.tokenType() != QXmlStreamReader::EndDocument);
+            DEBUG_ASSERT(reader.tokenType() != QXmlStreamReader::DTD);
+            DEBUG_ASSERT(reader.tokenType() != QXmlStreamReader::EntityReference);
+            // ignore other token types
```

Comment:
I think it is Characters, Comment and ProcessingInstructions.

Please name them explicit.

Reply:
Done